govcd/vapp_test.go

```diff
@@ -171,18 +171,46 @@ func (vcd *TestVCD) Test_SetOvf(check *C) {
 
 }
 
-// TODO: Add a check checking if the metadata was added to the vapp
-func (vcd *TestVCD) Test_AddMetadata(check *C) {
+func (vcd *TestVCD) Test_AddMetadataOnVapp(check *C) {
 	if vcd.skipVappTests {
 		check.Skip("Skipping test because vapp was not successfully created at setup")
 	}
-	// Add Metadata
+	// Add metadata
 	task, err := vcd.vapp.AddMetadata("key", "value")
 	check.Assert(err, IsNil)
 	err = task.WaitTaskCompletion()
 	check.Assert(err, IsNil)
 	check.Assert(task.Task.Status, Equals, "success")
 
+	// Check if metadata was added correctly
+	metadata, err := vcd.vapp.GetMetadata()
+	check.Assert(err, IsNil)
+	check.Assert(metadata.MetadataEntry[0].Key, Equals, "key")
+	check.Assert(metadata.MetadataEntry[0].TypedValue.Value, Equals, "value")
+}
+
+func (vcd *TestVCD) Test_DeleteMetadataOnVapp(check *C) {
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp was not successfully created at setup")
+	}
+	// Add metadata
+	task, err := vcd.vapp.AddMetadata("key2", "value2")
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
```

Comment:
This is too quick now: 
```
START: vapp_test.go:192: TestVCD.Test_DeleteMetadataOnVapp
vapp_test.go:199:
    check.Assert(task.Task.Status, Equals, "success")
... obtained string = "queued"
... expected string = "success"

FAIL: vapp_test.go:192: TestVCD.Test_DeleteMetadataOnVapp
```

```suggestion
  err = task.WaitTaskCompletion()
  check.Assert(err, IsNil)
	check.Assert(task.Task.Status, Equals, "success")
```

Reply:
Sorry for ugly suggestion formatting. The inline editor got me.

Reply:
Sure sorry did forget to push this. I added it and test are working.
```
START: vapp_test.go:192: TestVCD.Test_DeleteMetadataOnVapp
PASS: vapp_test.go:192: TestVCD.Test_DeleteMetadataOnVapp	9.688s

START: vm_test.go:596: TestVCD.Test_DeleteMetadataOnVm
Running: TestVCD.Test_DeleteMetadataOnVm
PASS: vm_test.go:596: TestVCD.Test_DeleteMetadataOnVm	10.449s
```